### PR TITLE
updates required for ruby 3.2.2

### PR DESCRIPTION
**Pull request by activemq**
Changes required for upgrading to ruby 3.2.2

### CODE REVIEW

File: test/admin-hpa/features/check_validator.feature [hunk 1]
```diff
@@ -21,6 +21,7 @@ Feature:
     Given a pupil has completed the check with the audit log is not contained in an array
     Then I should see an error stating validation failed as the audit log is not an array
 
+  @wip
```
Comment (jon-shipley) [inline test/admin-hpa/features/check_validator.feature:24]: What's the plan here?

File: test/pupil-hpa/features/support/sql_db_helper.rb [hunk 2]
```diff
@@ -463,4 +463,25 @@ def self.set_school_as_test_school(dfe_number)
     result = SQL_CLIENT.execute(sql)
     result.do
   end
+
+  def self.received_check(check_code)
+    sql = "SELECT * FROM [mtc_admin].[check] WHERE checkCode = '#{check_code}' and received=1"
```
Comment (jon-shipley) [inline test/pupil-hpa/features/support/sql_db_helper.rb:468]: This is insecure.  Please use parameters.

Comment (activemq) [reply test/pupil-hpa/features/support/sql_db_helper.rb:468]: @jon-shipley As discussed, we will look at using parameters in a future PR

Comment (GuyHarwood) [reply test/pupil-hpa/features/support/sql_db_helper.rb:468]: is there not an API for this in the `SQL_CLIENT`? it should be really trivial to add...

**Review by jon-shipley — APPROVED** [review]
LGTM

The parameterisation will need to be addresses in a separate PR.

**Pull request merged**